pyvisa-py/tcpip.py

```diff
@@ -401,56 +404,60 @@ def read(self, count):
         term_char, _ = self.get_attribute(constants.VI_ATTR_TERMCHAR)
         term_byte = common.int_to_byte(term_char) if term_char else b''
         term_char_en, _ = self.get_attribute(constants.VI_ATTR_TERMCHAR_EN)
-        timeout, _ = self.get_attribute(constants.VI_ATTR_TMO_VALUE)
-        timeout /= 1000.0
         suppress_end_en, _ = self.get_attribute(constants.VI_ATTR_SUPPRESS_END_EN)
 
         read_fun = self.interface.recv
 
-        out = bytearray()
-        out.extend(self._pending_buffer)
-
-        # minimum is in interval 1 - 100ms based on timeout
-        min_select_timeout = max(min(timeout/100.0, 0.1), 0.001)
+        # minimum is in interval 1 - 100ms based on timeout, 1sec if no timeut defined
+        min_select_timeout = 1 if self.timeout is None else max(min(self.timeout / 100.0, 0.1), 0.001)
         # initial 'select_timout' is half of timeout or max 2 secs (max blocking time).
         # min is from 'min_select_timeout'
-        select_timout = max(min(timeout/2.0, 2.0), min_select_timeout)
-        # time, when loop shall finish
-        finish_time = time.time() + timeout
+        select_timout = 2.0 if self.timeout is None else max(min(self.timeout / 2.0, 2.0), min_select_timeout)
+        # time, when loop shall finish, None means never ending story if no data arrives
+        finish_time = None if self.timeout is None else (time.time() + self.timeout)
         while True:
 
             # check, if we have any data received (from pending buffer or further reading)
-            if term_char_en and term_byte in out:
-                term_byte_index = out.index(term_byte) + 1
-                self._pending_buffer = out[term_byte_index:]
-                return bytes(out[:term_byte_index]), constants.StatusCode.success_termination_character_read
-
-            if len(out) >= count:
-                self._pending_buffer = out[count:]
-                return bytes(out[:count]), constants.StatusCode.success_max_count_read
+            if term_char_en and term_byte in self._pending_buffer:
+                term_byte_index = self._pending_buffer.index(term_byte) + 1
+                if term_byte_index > count:
```

Comment:
this check improves handling of case when requested less data then recevied and also terchar received. (#101)